### ISSUE TITLE
Crash when application context is passed to PaymentRequest

### ISSUE THREAD

**Issue by parakhod** [Adyen/adyen-android#28]
We had a crash in some situations after we integrated this module in our React Native app:

```
Fatal Exception: android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
       at android.app.ContextImpl.startActivity(ContextImpl.java:672)
       at android.app.ContextImpl.startActivity(ContextImpl.java:659)
       at android.content.ContextWrapper.startActivity(ContextWrapper.java:331)
       at android.content.ContextWrapper.startActivity(ContextWrapper.java:331)
       at com.adyen.ui.DefaultPaymentRequestListener.onPaymentDataRequested(DefaultPaymentRequestListener.java:34)
       at com.adyen.core.PaymentStateHandler.requestPaymentData(PaymentStateHandler.java:191)
       at com.adyen.core.PaymentStateHandler.onStateChanged(PaymentStateHandler.java:121)
       at com.adyen.core.internals.PaymentProcessorStateMachine.onStateChanged(PaymentProcessorStateMachine.java:26)
       at com.adyen.core.internals.PaymentProcessorStateMachine.onTrigger(PaymentProcessorStateMachine.java:39)
       at com.adyen.core.PaymentRequest.start(PaymentRequest.java:60)
...
```

After I changed React context passed to Payment request to getCurrentActivity() everything seems to work smooth (it's under testing now).
But anyway, it seems that it will be better to add `intent.setFlag(Intent.FLAG_ACTIVITY_NEW_TASK);` to the DefaultPaymentRequestListener.java before line 34, it would resolve the similar problems in the future.

**Comment by timon-langlotz** [comment]
Hi @parakhod,

first of all, sorry for the delay in the response. The `context` you pass in should indeed be an `Activity`. Setting `Intent.FLAG_ACTIVITY_NEW_TASK` is something we don't want to do (we don't want to actually start a new task here). Thanks for reporting!

**Comment by timon-langlotz** [comment]
Hey @parakhod,

we've just released version 2.0.0. This version does not have this issue.

Regards,
Timon
